<?xml version="1.0"?>
<launch>
  <arg name="paused" default="false"/>
  <arg name="gazebo_gui" default="true"/>
  <arg name="urdf_path" default="$(find hc10_ros)/urdf/hc10_only.urdf"/>

  <!-- startup simulated world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" default="worlds/empty.world"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gazebo_gui)"/>
  </include>

  <!-- send robot urdf to param server -->
  <param name="robot_description" textfile="$(arg urdf_path)" />

  <!-- push robot_description to factory and spawn robot in gazebo at the origin, change x,y,z arguments to spawn in a different position -->
  <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model motoman_hc10 -x 0 -y 0 -z 0"
    respawn="false" output="screen" />

  <!-- <include file="$(find motoman_hc10_moveit_config)/launch/ros_controllers.launch"/> -->

  <!-- loads the controllers -->
  <!--<rosparam file="$(find hc10_ros)/config/control.yaml" command="load" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" ns="/hc10_model" args="joint3_position_controller joint2_position_controller joint1_position_controller joint_state_controller"/> 
  
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
      <remap from="/joint_states" to="/simple_model/joint_states" />
</node> -->

</launch>
